source/client/stream_decoder.h

```diff
@@ -45,8 +45,8 @@ class StreamDecoder : public Envoy::Http::ResponseDecoder,
                 OperationCallback caller_completion_callback, Statistic& connect_statistic,
                 Statistic& latency_statistic, Statistic& response_header_sizes_statistic,
                 Statistic& response_body_sizes_statistic, Statistic& origin_latency_statistic,
-                HeaderMapPtr request_headers, bool measure_latencies, uint32_t request_body_size,
-                Envoy::Random::RandomGenerator& random_generator,
+                HeaderMapPtr request_headers, std::string request_body, bool measure_latencies,
```

Comment:
In a followup we should reevaluate the lifetime of the request body and consider string_view or string& parameters in various places.

Currently we are doing a very safe option, both passing strings by value and storing the value in the StreamDecoder, but there may be some avoidable copies that can carefully be eliminated. For now we can proceed as-is despite the small overhead of copying "".